src/middleware.js

```diff
@@ -16,17 +20,30 @@ const generateEditPageURL = (handle) => `${SITE_URL}/tickets/${handle}/edit`;
 export async function middleware(req) {
   try {
     const { pathname } = req.nextUrl;
-
-    // Exclude static files and API routes
-    if (
-      pathname.startsWith('/_next/') ||
-      pathname.startsWith('/favicon.ico') ||
-      pathname.startsWith('/api/')
-    ) {
-      return NextResponse.next();
+    
+    try {
+      const isLoggedIn = await checkCookie('neon_login_indicator');
+      if (pathname === '/' && isLoggedIn) {
+        try {
+          const referer = await getReferer();
+          if (
+            referer.includes(process.env.VERCEL_BRANCH_URL) ||
+            referer.includes(process.env.NEXT_PUBLIC_DEFAULT_SITE_URL)
+          ) {
+            return NextResponse.redirect(new URL(`${SITE_URL}/home`));
```

Comment:
Seems like this would work, but either:
- `new URL` is redundant here
- or let's use `new URL`'s 2nd parameter

so:
```suggestion
            return NextResponse.redirect(`${SITE_URL}/home`);
```

or:
```suggestion
            return NextResponse.redirect(new URL('/home', SITE_URL));
```

---

But actually [docs](https://nextjs.org/docs/app/building-your-application/routing/middleware#example) mention just using `req.url` as the base:

```suggestion
            return NextResponse.redirect(new URL('/home', req.url));
```